src/circuit/qc_keyagg_sw.rs

```diff
@@ -0,0 +1,262 @@
+//! Circuit implementation of stake key aggregation for quorum certificates verification.
+use ark_ec::{
+    short_weierstrass::{Projective, SWCurveConfig},
+    CurveGroup,
+};
+use ark_ff::PrimeField;
+use ark_std::{format, vec, Zero};
+use jf_primitives::{circuit::rescue::RescueNativeGadget, rescue::RescueParameter};
+use jf_relation::{
+    errors::CircuitError,
+    gadgets::{
+        ecc::emulated::{EmulatedSWPointVariable, SWPoint},
+        EmulationConfig,
+    },
+    BoolVar, Circuit, PlonkCircuit, Variable,
+};
+
+#[derive(Debug, Clone)]
+/// Stake public key variable
+/// Wrap EmulatedSWPointVariable because we need to simulate curve operations over the same curve's scalar field
+pub struct VerKeySWVar<E: PrimeField>(pub EmulatedSWPointVariable<E>);
+
+/// Plonk circuit gadget for stake key aggregation for quorum certificates.
+/// Assuming that the underlying key composes of a short Weierstrass curve point.
+pub trait QCKeyAggregateSWGadget<F>
```

Comment:
Can we have a unified trait for both SW points and TE points?

The implementations also seem pretty similar, is it possible to have a generic implementation?